lib/react.js

```diff
@@ -316,11 +316,24 @@ declare module react {
     initialState: (() => S) | S,
   ): [S, ((S => S) | S) => void];
 
+  declare type Dispatch<A> = (A) => void;
+
   declare export function useReducer<S, A>(
     reducer: (S, A) => S,
     initialState: S,
-    initialAction: ?A,
-  ): [S, A => void];
+  ): [S, Dispatch<A>];
+
+  declare export function useReducer<S, A>(
+    reducer: (S, A) => S,
+    initialState: S,
+    init: void,
+  ): [S, Dispatch<A>];
+
+  declare export function useReducer<S, I, A>(
```

Comment:
Is this the best ordering of the generics for when you specify them explicitly or would `<S, A, I>` be better so that the action type line up in the same place? Is there any practical difference?

Reply:
I don't think there's a practical difference, happy to align it. If I squint, I could make the argument that an initialArg would always 'occur' in source before any dispatch calls, so it's chronologically sound?

Reply:
It seems like the TypeScript defs need S and A to line up in a separate type for some obscure reason and therefore adds `I` at the end. So switching it would line up with TS.

Reply:
done and done